### ISSUE TITLE
sdl2.Texture.update() broken with transparent surfaces

### ISSUE THREAD

**Issue by NALStudio** [pygame-community/pygame-ce#2451]
**Environment:**

You can get some of this info from the text that pops up in the console when you run a pygame program.

- **Operating system**: Windows
- **Python version**: 3.11.5
- **SDL version**: 2.26.5
- **PyGame version**: 2.3.2

**Current behavior:**
When passing a transparent surface into sdl2.Texture.update(), the resulting texture becomes a blank white texture.

**Expected behavior:**
Load the image correctly, simple as that. Results should be same as images without transparency (expect for the transparency of course) and sdl2.Texture.from_surface() which handles transparency correctly.

**Screenshots**
<details><summary>Image WITHOUT transparency</summary>

![image](https://github.com/pygame-community/pygame-ce/assets/57502434/7fd52938-52b7-4581-88e5-25b838180c1b)
</details> 
<details><summary>Image WITH transparency</summary>

![image](https://github.com/pygame-community/pygame-ce/assets/57502434/61ae011e-d6b1-4c67-84ae-9afb0ab62ae1)
</details> 

**Steps to reproduce:**
1. Unzip the demo provided below
2. Install pygame-ce
3. Run _main.py

**Demo:**
[doesn't work.zip](https://github.com/pygame-community/pygame-ce/files/12623691/doesn.t.work.zip)

**Comment by yunline** [comment]
To temporarily fix this, use `Texture.from_surface()` instead.
```py
def load_tex(filepath: str) -> Texture:
    surf = pygame.image.load(filepath)
    tex = Texture.from_surface(rnd,surf)
    return tex
```

**Comment by NALStudio** [comment]
Yeah I'm already doing that, just decided to make a bug report as I'm not able to set the texture quality when using `Texture.from_surface()`

**Comment by zoldalma999** [comment]
Hey! Good to see another person trying out _sdl2!

Currently, while the surface format is converted if it is not the one the renderer uses in `.update`, it ignores any blend flags. You will need to explicitly set the `.blend` attribute of the texture to one of the `pygame.BLENDMODE_` flags (`BLENDMODE_BLEND` in this case). With that, it works fine for me. (This needs to happen on surfaces as well, this is what `convert_alpha` and the `pygame.SRCALPHA` flag do). So the `load_tex` function would look like this:
```py
def load_tex(filepath: str) -> Texture:
    surf = pygame.image.load(filepath)
    tex = Texture(rnd, surf.get_size(), static=True, scale_quality=2)
    tex.update(surf)
    tex.blend_mode = pygame.BLENDMODE_BLEND
    return tex
```


The reason it does work with `.from_surface` is because that *does* set this flag for you.

We should probably set it for ourselves as well, if the surface has alpha, or at the very least document that people need to do this.

**Comment by NALStudio** [comment]
Oh my god you are a legend @zoldalma999, thanks!
So is this intended behaviour? If it is, I'll close this issue then...

**Comment by zoldalma999** [comment]
I am happy that I could help.

Since this is an experimental module, where we can change things freely, there should be some discussion of what we do about this. But at least it should be documented that this does not set that flag, and that the user might need to. So I'll leave this open for now
